test/test_ops.py

```diff
@@ -71,6 +71,12 @@ def test_autocast(self, x_dtype, rois_dtype):
         with torch.cuda.amp.autocast():
             self.test_forward(torch.device("cuda"), contiguous=False, x_dtype=x_dtype, rois_dtype=rois_dtype)
 
+    @pytest.mark.parametrize('x_dtype', (torch.float, torch.half))
+    @pytest.mark.parametrize('rois_dtype', (torch.float, torch.half))
+    def test_autocast_cpu(self, x_dtype, rois_dtype):
```

Comment:
instead of creating a new test, maybe we could just parametrize over the device with the `cpu_and_gpu()` function?
Since the context manager is device-dependent, we could just set it in the code, like

```py
cm = torch.cpu.amp.autocast if device == 'cpu' else torch.cuda.amp.autocast
with cm():
    self.test_forward(torch.device(device), contiguous=False, x_dtype=x_dtype, rois_dtype=rois_dtype)
```

We could do the same for the rest of the newly introduced tests

Reply:
Yeah, good idea, I will modify it like this.